samples: nrf9160: modem_shell: kill button support (#5285)
nrfconnect/sdk-nrf: samples/nrf9160/modem_shell/README.rst

.. include:: /includes/build_and_run_nrf9160.txt

DK buttons
==========

The buttons have the following functions:

Button 1:

>     * Raises a kill or abort signal. A long press of the button will kill or abort all supported running commands. You can abort commands ``iperf3`` (also with ``th``), ``curl``, ``ping`` and ``sock send``.

Reviewer: No need for the bullets here. See, for example, https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples/bluetooth/central_nfc_pairing/README.html

Author: Bullets removed from here.

.. include:: /includes/build_and_run_nrf9160.txt

DK buttons
==========

The buttons have the following functions:

Button 1:

>     * Raises a kill/abort signal. I.e. long enough press of the button will kill/abort all supported running commands. The following commands can be aborted: iperf3 (also with th), curl, ping and sock send.

Reviewer: ```suggestion
    Raises a kill or abort signal. A long press of the button will kill or abort all supported running commands. You can abort commands ``iperf3`` (also with th), ``curl``, ``ping`` and ``sock send``.
```

Author: Updated.

Button 2:

>     * Enabling / disabling the UARTs for power consumption measurements. Toggles between UARTs enabled and disabled.

Reviewer: ```suggestion
    Enables or disables the UARTs for power consumption measurements. Toggles between UARTs enabled and disabled.
```

Author: Updated.

.. include:: /includes/build_and_run_nrf9160.txt

DK buttons
==========

The buttons have the following functions:

Button 1:

>     * Pressing of the button: raising a kill/abort signal. I.e. long enough press of the button will kill/abort all supported running commands.

Reviewer 2: I would add list of supported commands. Something like:
"The following commands can be aborted: iperf3, curl, sock send -l"

Author: Updated (+ping and th).

>     * Releasing of the button: resetting a kill/abort signal.

Reviewer 2: I'm thinking if this bullet should be removed. Does user need to know about this resetting? It's something that (s)he cannot avoid once (s)he presses the button, and the key event that happens is killing of the process.
I may not fully understand the resetting part so I'm not sure.
@tokangas Opinions?

Author: Updated.

Button 2:

>     * Enabling / disabling the UART of the MoSh.

Reviewer 3: I would suggest changing this a bit, because the implementation disables both UARTs. Also I think it's good to mention the use case, because it might not be obvious.
```suggestion
    * Enabling / disabling the UARTs for power consumption measurements.
```

Author: Updated.

>     * Enabling / disabling the UARTs for power consumption measurements.

Reviewer 2: ```suggestion
    * Enabling / disabling the UARTs for power consumption measurements. Toggles between UARTs enabled and disabled.
```

Reviewer 2: @hiltunent Can you confirm that this is about toggling UARTs between enabled and disabled?

Reviewer 4: yes, that is correct.

Author: Updated.
